### PR TITLE
Allow symbols in input name

### PR DESCRIPTION
**Pull request by Gerych1984**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌

Allow symbols like "-" or "_" in attribute name. For example - for dynamic forms where field stored in DB and as uuid/random string format

### CODE REVIEW

**Review by terabytesoftw — APPROVED** [review]

File: src/FormModelInterface.php
Comment (xepozz) [inline src/FormModelInterface.php:29]: It affects BC, can we postpone for a next major release?

Comment (Gerych1984) [reply src/FormModelInterface.php:29]: Yes. But without that it have psalm test error

File: tests/Helper/HtmlFormTest.php
Comment (xepozz) [inline tests/Helper/HtmlFormTest.php:88]: Move it to data provider please

File: src/FormModelInterface.php
Comment (xepozz) [reply src/FormModelInterface.php:29]: Could you please show it?

Comment (terabytesoftw) [reply src/FormModelInterface.php:29]: This package has no release so it's fine to add it to the interface.

Comment (Gerych1984) [reply src/FormModelInterface.php:29]: > Could you please show it?

Of course. That was happend after remove this method from [validator](https://github.com/yiisoft/validator/) 9 days ago

Снимок экрана от 2022-03-05 11-09-15

File: tests/Helper/HtmlFormTest.php
Comment (Gerych1984) [reply tests/Helper/HtmlFormTest.php]: Done

File: src/FormModelInterface.php
Comment (arogachev) [reply src/FormModelInterface.php:29]: Yes, it's a side effect of https://github.com/yiisoft/validator/pull/177. I already added it in #183, but adding in this or a separate PR is also fine. No worries for BC break since validator, form and even request model do not have a stable release yet.

File: tests/TestSupport/Form/DynamicFieldsForm.php
Comment (arogachev) [inline tests/TestSupport/Form/DynamicFieldsForm.php:21]: Why overriding this method? Isn't a parent implementation enough?

File: tests/TestSupport/Form/DynamicFieldsForm.php
Comment (arogachev) [inline tests/TestSupport/Form/DynamicFieldsForm.php:32]: Since `hasAttribute()` is presented here, it can be used instead.

File: tests/TestSupport/Form/DynamicFieldsForm.php
Comment (arogachev) [inline tests/TestSupport/Form/DynamicFieldsForm.php:41]: https://github.com/yiisoft/form/pull/184/files#r820079129

File: tests/TestSupport/Form/DynamicFieldsForm.php
Comment (arogachev) [inline tests/TestSupport/Form/DynamicFieldsForm.php:12]: In my opinion, this creates a certain confusion between attributes / fields.

File: src/FormModelInterface.php
Comment (arogachev) [reply src/FormModelInterface.php:29]: Created a separate PR to not wait until others are merged. https://github.com/yiisoft/form/pull/185

File: tests/Helper/HtmlFormTest.php
Comment (arogachev) [inline tests/Helper/HtmlFormTest.php:102]: I suggest to use the classic way for working with data providers here. https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers

File: tests/TestSupport/Form/DynamicFieldsForm.php
Comment (arogachev) [reply tests/TestSupport/Form/DynamicFieldsForm.php:12]: How it was resolved? I don's see neither a comment nor a fix. @Gerych1984 @terabytesoftw

**Pull request merged**
